tests/test_graph_utils.py

```diff
@@ -507,3 +509,103 @@ def test_result_equality(
 
         assert all(
             repeat(next(first_factory) == next(second_factory), nb_steps))  # type: ignore
+
+
+class TestSubgraphSampling:
+
+    @staticmethod
+    def get_graph(rng):
+        """
+        Randomly create somewhat complex graphs, then return a complex one
+        """
+        factory = GraphRandomFactory(
+            [0.15, 0.15, 0.15, 0.15, 0.14],
+            10, 0.15, rng)
+        # go through generation steps
+        for _ in range(5):
+            next(factory)  # type: ignore
+        return factory.graphs[0]
+
+    def test_sample(self):
+        """ Test sample respect given probability distribution """
+        pass
+
+    def test_uniform(self):
+        """ Sanity checks on uniform sampler """
+        rng = random.Random()
+        g = self.get_graph(rng)
+        n_vertices = max(1, len(g) - 4)
+        sg = uniform_sample(g, n_vertices, rng)
+        assert 1 <= len(sg) <= n_vertices
+        assert all(v in g for v in sg)
+
+    # Sometimes the RandomFactory fails (erode) or the algorithm does not converge
```

Comment:
So this still happens? (RandomFactory failure). I thought your fix would have made this go away, too bad :'(

Reply:
Nope, leftover comment. To be removed